test/integration/scheduler_perf/util.go

```diff
@@ -202,22 +204,37 @@ func (*metricsCollector) run(ctx context.Context) {
 
 func (pc *metricsCollector) collect() []DataItem {
 	var dataItems []DataItem
-	for _, metric := range pc.Metrics {
-		dataItem := collectHistogramVec(metric, pc.labels)
-		if dataItem != nil {
-			dataItems = append(dataItems, *dataItem)
+	for name, labelPair := range pc.Metrics {
+		metricFamily, err := testutil.GetMetricFamilyFromGatherer(legacyregistry.DefaultGatherer, name)
+		if err != nil {
+			klog.Error(err)
+			return dataItems
+		}
+		// no filter is specified, aggregate all the metrics within the same metricFamily.
+		if labelPair == nil {
+			dataItem := collectHistogramData(metricFamily, pc.labels, nil, name)
+			if dataItem != nil {
+				dataItems = append(dataItems, *dataItem)
+			}
+		}
+		// build the filter by the pair of "label=value".
+		for label, values := range labelPair {
+			for _, value := range values {
+				dataItem := collectHistogramData(metricFamily, pc.labels, map[string]string{label: value}, name)
```

Comment:
I think you should compose a full label->value map, and then call `collectHistogramData`; instead of call it on every label->value pair.

For example, if the labelPair is:

```
		Metrics: map[string]map[string][]string{
			"scheduler_framework_extension_point_duration_seconds": {
				"extension_point": {"Filter", "Score"},
				"status":          {"0"},
			},
```

The current logic would call `collectHistogramData` on {extension_point:Filter}, {extension_point:Score} and {status:0}. However, the desired results is {extension_point:Filter, status:0} and {extension_point:Score, status:0}.

Reply:
And ensure the above-mentioned case is UTed.

Reply:
yes, this is a little complicated to handle, so I add a func of `descartes` to get all the combination,  updated `TestGetHistogramVecFromGatherer` only, pls let me know whether `descartes` should be UTed.

Reply:
> this is a little complicated to handle

To support "multi-filtering", it seems the current logic is a bit too complicated. I'm a bit skeptical about if the benefits could outweigh the maintenance efforts...

Bear with me for the comments back and forth, I'm now more inclined to restrict the metric to a single filter, which means to change the `Metrics` struct like:

```go
metrics: map[string]*labelValues

type labelVals struct {
    label string
    values []string
}
```

WDYT?

Reply:
> Bear with me for the comments back and forth

No worries, not just one answer and we are all trying to find the best one.